lms/static/scripts/postmessage_json_rpc/client/client.js

```diff
@@ -0,0 +1,98 @@
+import { generateHexString } from './random';
+
+/** Generate a random ID to associate RPC requests and responses. */
+function generateId() {
+  return generateHexString(10);
+}
+
+/**
+ * Return a Promise that rejects with an error after `delay` ms.
+ */
+function createTimeout(delay, message) {
+  return new Promise((_, reject) => {
+    setTimeout(() => reject(new Error(message)), delay);
+  });
+}
+
+/**
+ * Make a JSON-RPC call to a server in another frame using `postMessage`.
+ *
+ * @param {Window} frame - Frame to send call to
+ * @param {string} origin - Origin filter for `window.postMessage` call
+ * @param {string} method - Name of the JSON-RPC method
+ * @param {any[]} params - Parameters of the JSON-RPC method
+ * @param [number] timeout - Maximum time to wait in ms
+ * @param [Window] window_ - Test seam.
+ * @param [id] id - Test seam.
+ * @return {Promise<any>} - A Promise for the response to the call
+ */
+async function call(
+  frame,
+  origin,
+  method,
+  params = [],
+  timeout = 2000,
+  window_ = window,
+  id = generateId()
+) {
+  // Send RPC request.
+  const request = {
+    jsonrpc: '2.0',
+    method,
+    params,
+    id,
+  };
+
+  try {
+    frame.postMessage(request, origin);
+  } catch (err) {
+    return Promise.reject(err);
```

Comment:
So it shows up in the console as an error. Not sure any of that should be user facing -- as an improvement we could add in a toast style notification to the UI.  I could also just wrap the `rcpCall` in `BasicLtiLaucnApp` in a try/catch and then formally print errors to `console.error`. That way it won't show up as an unresolved promise, but either way it does get to the console.

Reply:
Since this is now an async function, you can remove the `try..catch` block entirely as async/await automatically converts exceptions to rejected promises.